crates/cli/src/benchmark.rs

```diff
@@ -1,19 +1,128 @@
 use anyhow::{Context, Result};
+use rand::{rngs::SmallRng, Rng, SeedableRng};
 use sightglass_recorder::{benchmark::benchmark, measure::MeasureType};
-use std::ffi::OsString;
-use std::fs;
-use std::io;
-use std::path::PathBuf;
 use std::str::FromStr;
+use std::{fmt, io};
+use std::{fs, process::Command};
+use std::{path::PathBuf, process::Stdio};
 use structopt::StructOpt;
 
-/// Measure the compilation, instantiation, and execution of a Wasm file.
+/// Measure compilation, instantiation, and execution of a Wasm file.
+///
+/// The total number of samples taken for each Wasm benchmark is `PROCESSES *
+/// SAMPLES_PER_PROCESS`.
 #[derive(StructOpt, Debug)]
-#[structopt(name = "benchmark")]
 pub struct BenchmarkCommand {
+    /// The path to the shared library implementing the benchmark API.
+    #[structopt(
+        short("e"),
+        long("engine"),
+        value_name = "ENGINE",
+        parse(from_os_str),
+        empty_values(false)
+    )]
+    engines: Vec<PathBuf>,
+
+    /// How many processes should we use for each Wasm benchmark?
+    #[structopt(long = "processes", default_value = "30", value_name = "PROCESSES")]
+    processes: usize,
+
+    /// How many samples should we take from a single process?
+    #[structopt(
+        long = "samples-per-process",
+        default_value = "1",
+        value_name = "SAMPLES_PER_PROCESS"
+    )]
+    samples_per_process: usize,
```

Comment:
Some thoughts about this: 
 - should the in-process and multi-process benchmark commands be the same struct? (The in-process struct could use the same `Vec<_>` parameters for engines and Wasm files)
 - if they should be separate structs, should we call this one `MultiProcessBenchmarkCommand`?
 - how are we triggering this one? The other one corresponds to the `benchmark` command... is structopt somehow smart enough to see the extra parameters and choose this struct?
 - I didn't think we needed to spell out the `short = ...` and `long = ...` attributes?

Reply:
The `benchmark` subcommand is the multi-process runner now. The idea is that users basically won't use `in-process-benchmark`, just the regular `benchmark` subcommand will. We might even want to hide it from the help text.

I don't think we want to make the in-process benchmark commands the same. The struct definition contains some duplication, but that isn't a maintenance burden. There isn't any duplicated code, which is where things would get annoying. And I expect that as we flush out the benchmark runner, they will diverge even further.

Reply:
I see the `#[structopt(name = "benchmark")]` on `InProcessBenchmarkCommand`?

Reply:
Ah woops! That was a leftover and/or rebase issue. Funnily enough, it didn't seem to have an effect, as the command was still showing up as `in-process-benchmark`. Removing it now.